weldx/core.py

```diff
@@ -357,24 +373,32 @@ def _initialize_discrete(
         if interpolation is None:
             interpolation = "step"
 
-        # expand dim for scalar input
-        data = Q_(data)
-        if not np.iterable(data):
-            data = np.expand_dims(data, 0)
-
-        # constant value case
-        if time is None:
-            time = pd.TimedeltaIndex([0])
-
-        if isinstance(time, pint.Quantity):
-            time = ut.to_pandas_time_index(time)
-        if not isinstance(time, pd.TimedeltaIndex):
-            raise ValueError(
-                '"time" must be a time quantity or a "pandas.TimedeltaIndex".'
-            )
+        if isinstance(data, xr.DataArray):
+            self._check_data_array(data)
+            if data.dims[0] != "time":
+                data = data.transpose(
+                    *ut.swap_list_items(data.dims, data.dims[0], "time")
+                )
+            self._data = data
```

Comment:
can't look up the code right now but there is a syntax that should work like:
~~~python
self. _data = data.transpose(..., "time") 
~~~
(check some LCS code)

Reply:
works... good to know

Reply:
I think the if check is not necessary with this